addons/bus/static/tests/im_status.test.js

```diff
@@ -1,126 +1,114 @@
     serverState,
-    step,
+    waitForSteps,
 } from "@web/../tests/web_test_helpers";
 import { defineBusModels } from "./bus_test_helpers";
-import { advanceTime, tick } from "@odoo/hoot-mock";
-import { browser } from "@web/core/browser/browser";
 
 defineBusModels();
 
-// Delays can slighly differ since time is not frozen. Let's tolerate 1000ms
-// of difference.
-const TOLERANCE = 1000;
-const AWAY_DELAY = ACTUAL_AWAY_DELAY + TOLERANCE;
-function expectAlmostEqual(a, b) {
-    expect(Math.abs(a - b) < TOLERANCE).toBe(true, {
-        message: `${a} and ${b} should be almost equal`,
-    });
-}
+beforeEach(freezeTime);
+
+describe.current.tags("headless");
 
 test("update presence if IM status changes to offline while this device is online", async () => {
+    mockService("bus_service", { send: (type) => asyncStep(type) });
+
     const { env } = await makeMockServer();
     await makeMockEnv();
-    patchWithCleanup(getService("bus_service"), { send: (type) => step(type) });
-    await assertSteps(["update_presence"]);
+    await waitForSteps(["update_presence"]);
     env["bus.bus"]._sendone(serverState.partnerId, "bus.bus/im_status_updated", {
         im_status: "offline",
         partner_id: serverState.partnerId,
     });
-    await assertSteps(["update_presence"]);
+    await waitForSteps(["update_presence"]);
 });
 
 test("update presence if IM status changes to away while this device is online", async () => {
+    mockService("bus_service", { send: (type) => asyncStep(type) });
+
+    localStorage.setItem("presence.lastPresence", Date.now());
+
     const { env } = await makeMockServer();
     await makeMockEnv();
-    patchWithCleanup(getService("bus_service"), { send: (type) => step(type) });
-    patchWithCleanup(getService("presence"), { getLastPresence: () => new Date().getTime() });
-    await assertSteps(["update_presence"]);
+    await waitForSteps(["update_presence"]);
     env["bus.bus"]._sendone(serverState.partnerId, "bus.bus/im_status_updated", {
         im_status: "away",
         partner_id: serverState.partnerId,
     });
-    await assertSteps(["update_presence"]);
+    await waitForSteps(["update_presence"]);
 });
 
 test("do not update presence if IM status changes to away while this device is away", async () => {
+    mockService("bus_service", { send: (type) => asyncStep(type) });
+
+    localStorage.setItem("presence.lastPresence", Date.now() - AWAY_DELAY);
+
     const { env } = await makeMockServer();
     await makeMockEnv();
-    patchWithCleanup(getService("bus_service"), { send: (type) => step(type) });
-    patchWithCleanup(getService("presence"), {
-        getLastPresence: () => new Date().getTime() - AWAY_DELAY,
-    });
-    await assertSteps(["update_presence"]);
+    await waitForSteps(["update_presence"]);
     env["bus.bus"]._sendone(serverState.partnerId, "bus.bus/im_status_updated", {
         im_status: "away",
         partner_id: serverState.partnerId,
     });
-    await tick();
```

Comment:
Hello, all of the ticks or other waiting before empty `waitForSteps` have to be kept otherwise they will resolve immediately, effectively not checking anything and losing coverage (unless the API of the function has changed, which doesn't seem to be the case).